### ISSUE TITLE
dotenvx set --plain does not escape values

### ISSUE THREAD

**Issue by coolhome** [dotenvx/dotenvx#378]
dotenvx set foo '$bar$baz$paz$1234' --plain

// Creates file with these lines:
// Line 1: foo="$bar$baz$paz
// Line 2: 234"

dotenvx get foo
// Empty because broken env file

**Comment by motdotla** [comment]
i can't repeat this. what shell are you in?

(this can get tricky when using set because your shell might try to interpret some of these values before dotenvx set has a chance to run. for example $1)

**Comment by motdotla** [comment]
if you escape the dollar signs, do you experience the same problem?

```
dotenvx set foo "\$bar\$baz\$paz\$1234" --plain
```

**Comment by motdotla** [comment]
i cannot reproduce this. maybe try upgrading your dotenvx. using `dotenvx@1.18.01`

```sh
$ dotenvx set foo '$bar$baz$paz$1234' --plain
```

results in:

```ini
# .env
foo='$bar$baz$paz$1234'
```

when using `dotenvx get` **correctly** returns an empty string because of  variable expansion

```sh
$ dotenvx get
{"foo":""}
```

if you don't want variable expansion escape your dollar signs:

```
$ dotenvx set foo '\$bar\$baz\$paz\$1234' --plain
$ dotenvx get
```

<img width="941" alt="image" src="https://github.com/user-attachments/assets/50a7c4de-a252-4654-9204-f49fe9172fe2">


hmm, that works on the first go around but not on the second. looks like there is an issue when `foo=` already exists in the .env file. looking into that.

**Comment by motdotla** [comment]
fixed in `v1.18.1`. was related to user inputted regex groups (`$1`) that node `replace` was attempting to interpolate.

<img width="633" alt="image" src="https://github.com/user-attachments/assets/f4585635-78d0-4151-aa7a-c101f8bd3e43">


thanks for finding this edge case!
